internal/dag/dag.go

```diff
@@ -351,18 +356,28 @@ func (d *DAG) recompute() dag {
 	}
 
 	// process ingressroute documents
+	var status []ingressrouteStatus
+	orphaned := make(map[meta]bool)
 	for _, ir := range d.ingressroutes {
 		if ir.Spec.VirtualHost == nil {
-			// delegate ingressroute, skip it
+			// delegate ingress route. mark as orphaned if we haven't reached it before.
```

Comment:
I don't think this needs a conditional. The logic that is here now says

_if not found, then mark as found_

which i think can be simplified into
```
orphaned[meta{name: ir.Name, namespace: ir.Namespace}] = true
```

Reply:
The problem is that we can overwrite a `false` that is already there. For example, in a scenario where A delegates to B, assuming we process A before B, we would reach B from A and mark it as orphaned = false. Once we are done processing A, we move onto processing B, and we don't want to overwrite orphaned = false.